### ISSUE TITLE
Redesign the Snapshot submission page from submitting to displaying submitted snapshots 

### ISSUE THREAD

**Issue by NilakshiS** [hackforla/tdm-calculator#2004]
### Overview
We need to update the Submit Snapshot page so that it only shows snapshots that have already submitted.

#### Change Page Title
The title of the page is being changed from "Submit Snapshot" to "Submissions".

#### Remove Tables
The following tables are being removed:
- Snapshot Meeting Target Points
- Snapshots Not Meeting Target Points

#### Update Aesthetics
The design of the Submissions page must be updated to align with the table style developed as part of #2013.

#### Alter Current Columns
- The "Date Modified" column will be removed.
- The "Status Updated" column will be added. It shows the most recent date upon which any of the fields in the below "Add New Columns" section were updated by an admin.

#### Add New Columns
All of the columns from the Snapshot Admin page detailed in #1953 will also be added to this page, with the exception of Created By, Admin Notes, Save, Reset, and Audit Log. Those columns include:
- **Level**: A read-only text field that will be auto-populated by the Project Level number.
- **DRO**: A dropdown that will include the following options.
  - [null]
  - Metro
  - Valley
  - West L.A.
- **Staff Assigned**: A dropdown that will include a list of all the admins in the TDM system. Users are given Admin status on the Security page.
- **Assigned Date** *: This field will only appear if the Staff Assigned field has data entered into it. If the user attempts to save when the Staff Assigned field has data entered into it but the Assigned Date field does not, a warning modal will appear, telling the user they must enter data in the Assigned Date field before they can save.
- **Invoice Status**: A dropdown that will include the following options.
  - Not sent
  - Sent
  - Paid
- **Invoice Paid Date** *: This field will only appear if the Invoice Status field has “Paid” selected in its dropdown. If the user attempts to save when the Invoice Status field has “Paid” selected but the Invoice Paid Date field has no data entered into it, a warning modal will appear, telling the user they must enter data in the Invoice Paid Date field before they can save.
- **On Hold**: A dropdown that will include the following options.
  - No
  - Yes
- **Approval Status**: A dropdown that will include the following options.
  - Pending
  - Terminated
  - Withdrawn
  - Denied
  - Approved
  - Approved with Conditions
- **Admin Notes**: A text entry field that shares its data with the Admin Notes field on the My Projects page. Any edits made to the field on the My Projects page are reflected to the same field on the Snapshot Admin page, and vice-versa.
- **Certificate of Occupancy Date**: A date field that will use the same date picker component as found in filters for date fields on the My Projects page.

*A date field that will use the same date picker component as found in filters for date fields on the My Projects page.

### Action Items
- [x] Create a slide for this issue on the [Staging Deck](https://docs.google.com/presentation/d/1crZ3IxqA4hAu3qzD7ns93Ieuqjwh6wyEtuX_46cP-fg).
- [x] Wait for #2013 to be completed.
- [x] Add a new page to the Design section of the Figma file and include this issue's number, title, and assignee in the title.
- [x] Use the Figma page to design solutions for this issue.
- [x] Review proposed designs with the design lead. Implement feedback as necessary.
- [x] Use [presentation structure](https://github.com/hackforla/tdm-calculator/wiki/Design:-Presentations) to review proposed designs with the design director. Implement feedback as necessary.
- [x] Use [presentation structure](https://github.com/hackforla/tdm-calculator/wiki/Design:-Presentations) to review proposed designs with the product, development, and research teams. Implement feedback as necessary.
- [x] Receive design approval from stakeholders. Implement feedback as necessary.
- [x] Add any annotations to the Figma design that would help the developer with implementation.
- [x] Create a Ready for Dev section around the final designs on the Figma page.
- [x] Update the [Figma Design System](https://www.figma.com/design/nD9QK56Mzq7xNSaSUoeGx0/TDM-Calculator?node-id=4041-12655&t=54f7L4Eubz7ppygs-1) with changes resulting from the issue's final design, if applicable.
- [ ] Update the [Style Guide Presentation](https://docs.google.com/presentation/d/1I4q35NL2WW2RpksIyawhHCd8qJnrjme1ZDNIBxu24hQ/), if applicable.
- [x] Add a comment to the issue that references the final design and adds any annotations that would help the developer with implementation.
- [x] Create a [development handoff issue](https://github.com/hackforla/tdm-calculator/issues/new?template=development-handoff-issue.md) that references the comment.
- [x] Move the Figma page to the Development section of the Figma file.
- [x] Add the development issue's number in parentheses at the beginning of the title of the Figma page.
- [x] Include the development handoff issue number in the GitHub section below.
- [x] Reassign any previous assignees to this issue.
- [x] Close this issue.

### Resources/Instructions
- #1407
- #1655
- #1953 
- #2013  
- [Link to the Design System on Figma](https://www.figma.com/design/nD9QK56Mzq7xNSaSUoeGx0/TDM-Calculator?node-id=17182-64787&t=ymjepbPzUO3SeV1A-1)
- [Link to Submit Snapshot on TDM Calculator Dev site](https://tdm-dev.azurewebsites.net/submit)
- Development handoff issue: #2372

**Comment by ExperimentsInHonesty** [comment]
This is the comment on a prior issue where we changed from check boxes to radio buttons. 
  They can only submit one project at a time. https://github.com/hackforla/tdm-calculator/issues/1407#issuecomment-1828907896

**Comment by Parisajf** [comment]
Combine the two charts (meeting the target point, not meeting the target point)
The bottom chart if they are not active will have a gray header and the active chart will have a navy blue header.
We don’t want to submit anything in this page, instead we want  to submit through “submitted projects “ page, we will need to have a note there that for advanced filtering see status in my project’s page.  Also In “my projects” Page, the non admin users will see 1. Target points 2. Earned points for their own projects. We wont need to duplicate it in the submit form.

**Comment by greenishviolet** [comment]
### 2/18 Progress

#### Progress
- I created the first iteration of the Submission Page

![Image](https://github.com/user-attachments/assets/62384927-16f1-49d7-85f7-faf749b32d72)

- I made iterations of the subheader:
     - **More direct:** Submit your snapshots through My Projects or Page 5 of your project.
     - **More urgent:** You must submit your snapshots in My Projects or on Page 5 of your project.
     - **More concise:** Submit snapshots via My Projects or on Page 5 of your project.
     - **More explicit/softer tone:** You can submit your snapshot in My Projects or on Page 5 of your project.

- Currently, table headers have filter icons, where users can sort items in the table. I haven't had the time to find a more appropriate icon or go through past issues to see if we have already decided on icons. I will look into that for next week.
     - However, this makes me wonder if the Submissions table will have the same filtering and sorting functionality as the My Projects table (and if that is feasible given the timeline we have).

- I also played around with reordering menu items in the navigation so links are shown in the following order: **Create Project** → **My Projects** → **Submissions**
     - I felt that putting **Submissions** after **My Projects** will make navigation more intuitive, as users can easily access My Projects if they want to submit a snapshot.
     - However, this could be confusing to users since Submissions and My Projects sound like 2 similar things.

![Image](https://github.com/user-attachments/assets/0d0aaaa0-f6ce-4114-8222-6f5b892ae0e7)

#### Blockers
- I would like more context about sorting in my submissions. Will sorting for the Submissions table work like sorting in the My Projects table (aka. will an overlay open to show users sorting options?)?

<details>
<summary>Filtering overlay in My Projects</summary>

![Image](https://github.com/user-attachments/assets/1ebc49c0-2f7d-4101-be08-bfdeb3707c06)

</details>

#### Availability
I have around 6 hours to work on this issue next week

#### ETA
Probably in 2 or 3 weeks?

**Comment by thetanmancan** [comment]
@greenishviolet Thank you for this update. I will move this issue to the Questions/Clarifications column so that we can get an answer to your question about whether or not sorting will be accessed via filter overlays in the same way that they are on the My Projects page.

**Comment by thetanmancan** [comment]
@greenishviolet, per [this comment](https://github.com/hackforla/tdm-calculator/issues/2013#issuecomment-2675954494), I'm having @namigoeku reconcile the addition of the new column being added on the My Projects page via [this comment](https://github.com/hackforla/tdm-calculator/issues/2004#issuecomment-2667252504). If you have any questions about it, please contact her or me. Thank you.

**Comment by thetanmancan** [comment]
I'm going to leave this issue in the Questions/Clarifications column so that we can get an answer from the production and development teams in regards to the prior question:
- If the Submissions page will only have sorting functionality, will it work the same way as on the My Projects page, where the user clicks the filter icon and opens an overlay and selects a radio button and clicks Apply? Or will there just be a sort icon that the user clicks?

**Comment by ExperimentsInHonesty** [comment]
@thetanmancan @greenishviolet @entrotech 

>Q: I'm going to leave this issue in the Questions/Clarifications column so that we can get an answer from the production and development teams in regards to the prior question:

If the Submissions page will only have sorting functionality, will it work the same way as on the My Projects page, where the user clicks the filter icon and opens an overlay and selects a radio button and clicks Apply? Or will there just be a sort icon that the user clicks?

A: My recommendation is that the column just sorts by icon (up down arrows) but it might be less coding to just copy the filters from the my project's page for those fields.  John, what do you think?

**Comment by greenishviolet** [comment]
### 2/25 Progress

#### Progress
I have not been able to work on this issue this week.

#### Blockers
Currently, none.

#### Availability
I think I am able to give 4 hours to this issue. I will try to create a mockup next week.

#### ETA
Probably 2 weeks

#### Pictures
N/A

**Comment by thetanmancan** [comment]
@greenishviolet, I wanted to make you aware of some specifications that came from the product and development teams from our 2025-02-26 meeting.
- The same filtering and sorting features in the column headers from the My Projects page—in other words, the functionalities in the overlays—need to be applied to the Submissions page table.
- The Reset Filters/Sort button will also be included on the Submissions page.
- There will not be a search bar on the Submissions page.

Let me know if you have any questions or concerns. Thanks.

**Comment by thetanmancan** [comment]
Hi @greenishviolet. When you get a chance, can you please provide an update for this issue? Thank you.

**Comment by greenishviolet** [comment]
### 03/04 Progress

#### Progress
I made a new iteration of the page:
- Removed the sidebar so users can focus on the table
- Updated the icon for sorting in table headers
     - I'm working with @namigoeku to figure out icons. I have shared mockups of icons in different states with her.
- I have also applied the specifications @thetanmancan brought up two days ago.
- I also edited the overlays from My Projects to be used for sorting

I have yet to update the copy

#### Blockers
There's an issue with icons that I wanted to bring up. In #1816, Tristan used the rotated **switch-right** icon to show that sorting was applied on the table.

Here's an image for reference:

![Image](https://github.com/user-attachments/assets/49a8049b-4b63-4206-9179-08071ef9c416)

Material design does not have an icon where none of the triangles are filled in. This icon would be used to denote that sorting has not been applied, and to open overlays to apply sorting.

This icon would look like this:

![Image](https://github.com/user-attachments/assets/9f199a1d-38a7-4aec-a78b-bdab0e4be4e5)

I created this icon using the **switch-right** icon from Material Design. I did some research and saw that a similar icon is available on [Font Awesome](https://fontawesome.com/icons/sort?f=classic&s=regular), except that you need the Font Awesome Pro to use this. I'm not sure if a pro account is necessary.

#### Availability
I have 5-6 hours to work on this issue this week.

#### ETA
I think I need 2 more weeks for this issue.

#### Pictures
<details>
<summary>Iteration 2</summary>

![Image](https://github.com/user-attachments/assets/28a82704-ba5b-43ec-9221-22ef42718761)

</details>

<details>
<summary>Overlays</summary>

![Image](https://github.com/user-attachments/assets/24ade594-1089-4590-9dd5-35c4c3db848a)

</details>

<details>
<summary>Icons</summary>

![Image](https://github.com/user-attachments/assets/0b7656ba-12b7-4402-8895-731ec8ea8456)

</details>

**Comment by thetanmancan** [comment]
Hi @greenishviolet. I took a look at your updates and have these points to make:
- In principle, we're meant to leave the sidebar on all pages that aren't the My Projects page for aesthetic reasons. If you find that the table isn't fitting on the Submissions page, please let me know and we can run the idea of removing the sidebar by the product team, but otherwise please plan to keep it in.
- There is no need for a double-empty chevron icon because the filtering and sorting will work exactly the same way it does on the My Projects page. On the My Projects page, the sort icon only appears if sorting is active for that column.
- The filter overlays will still need a search function and a combo box or date picker like they do on the My Projects page. I assume I wasn't being clear in [this comment](https://github.com/hackforla/tdm-calculator/issues/2004#issuecomment-2693004390) when I said that there wouldn't be a search bar. My mistake. What I meant was that there won't be a search bar like the one above the table on the My Projects page. 
- What you have as the "Reset Sort" button should read as “Reset Filters/Sort”.
- In the subheader text, there should be a sentence that explains what the purpose of the page is before explaining where the user can submit their snapshots.

Thank you for your work this week. Please let me know if you have any questions.

**Comment by greenishviolet** [comment]
### 3/11 Update

#### Progress
- I created a new mockup of the Submissions page, taking feedback from @thetanmancan into consideration.
     - I added the sidebar back in
     - I updated label in the Reset button
     - I replaced the sort icon with the filter icons form #1816
- In addition, I found the filter and sort overlays from #1816 for Name, Address, and Date Submitted/Modified
- I also made iterations on the subheader text

#### Availability
I have around 4 hours to work on this issue

#### ETA
I think I might need 1 more week to make edits based on feedback I receive during the week.

#### Pictures
<details>
<summary>Mockup of Submissions page</summary>

![Image](https://github.com/user-attachments/assets/4e8f634d-fd15-4150-bf4c-ca4d23672531)

</details>

<details>
<summary>Filter Overlays</summary>

![Image](https://github.com/user-attachments/assets/9262968a-3c96-42a5-a7a6-82c1bb612525)

![Image](https://github.com/user-attachments/assets/c0aa04cc-9691-43cc-af7b-d88f7e587ed9)

![Image](https://github.com/user-attachments/assets/0b3a81a3-c4cd-473b-9c25-27f24d2eb82e)

</details>

<details>
<summary>Copy Iterations</summary>

![Image](https://github.com/user-attachments/assets/7fe800f7-bdd1-4fd7-8f0a-5f77e5c05ca8)

</details>

**Comment by thetanmancan** [comment]
@greenishviolet, please make sure the date overlays receive the same drop shadow as the other overlays. Thank you.

**Comment by greenishviolet** [comment]
@thetanmancan, I have updated the drop shadow styles to the date overlays.

![Image](https://github.com/user-attachments/assets/69ce387d-1390-48c2-9299-bb48712574ad)

**Comment by thetanmancan** [comment]
Thank you, @greenishviolet.

Per today's prep meeting, Bonnie asked for a few more changes.
- In the subheader, "My Projects" should be a link.
- The snapshot names in the table, i.e. "Blythe Street", should be links.
- There should be language somewhere on the page that lets the user know that they can get more information about their projects on the My Projects page. Bonnie suggested putting that on the top-left of the table, but you may want to consider other options.

It'd be preferable if you could get those changes done by Tuesday's stakeholder meeting, but if you can't, no biggie. If you have any questions or concerns about any of those changes, please let me know. Thank you.

**Comment by greenishviolet** [comment]
@thetanmancan - I updated the table with changes as you requested.

![Image](https://github.com/user-attachments/assets/f3985038-908f-4c08-9f6f-2020048d29ee)

I added the language to the subheader. It makes sense for user to find information about where to find more information about a Project since instructions on where to go to submit a project is already in the subheader.

I also have two other options for copy. Let me know if you want me to swap copy or write something else.

![Image](https://github.com/user-attachments/assets/aac9ce3b-7d99-45f5-97e7-41419c2bf8fb)

**Comment by greenishviolet** [comment]
### 3/18 Progress

#### Progress
I have not worked on this issue since the last update

#### Availability
I have around 4 hours to work on this issue

#### ETA
Possibly 1 more week, depends on what was decided in the stakeholder meeting.

#### Pics
N/A

**Comment by greenishviolet** [comment]
Hi @ExperimentsInHonesty, do you mean updates based on feedback from the stakeholder meeting? @thetanmancan told me that he will let me know what the next steps are on this issue tomorrow.

**Comment by thetanmancan** [comment]
@greenishviolet, here are the notes from the 2025-03-18 meeting.
- Your proposal to change the order of the navigation bar to "Create Project, My Projects, Submissions, About, FAQ, Feedback" was approved by stakeholders. They also asked that you incorporate the Snapshot Admin, Admin Guide, and Security pages to that ordering, depending on whether a user, admin, or security admin is logged in. Please reorder the navigation bar, taking into account those three additional pages and the type of logged in user.
- In response to #2195, stakeholders asked that the fields being added to the Snapshot Admin page via #1953 be added to the Submissions page, with the exception of the Admin Notes, Save, Reset, and Audit Log fields and buttons.
  - GitHub issue: #1953 
  - Nilakshi's [Figma page](https://www.figma.com/design/nD9QK56Mzq7xNSaSUoeGx0/TDM-Calculator?node-id=18525-24&t=72EubOwft1XFeDcf-1)
  - Please remove the sidebar that you were originally asked to include, since the page will definitely no longer have room for it.
  - Please include the fields that are already on the mockup for the Submissions page, i.e. Name, Address, Date Submitted, and Date Modified.
- As a heads up, the "Date Modified" label will have its name changed. We are going to ask stakeholders what they would like its name to be changed to.

I know this is a lot you're being asked to do. Please feel free to reach out to me with any questions you may have. Thank you.

**Comment by thetanmancan** [comment]
@greenishviolet, in response to the questions you sent me:

## ✅ Clarification of Issue #2004 Requirements

### Updating the navigation bar

> "They also asked that you incorporate the Snapshot Admin, Admin Guide, and Security pages to that ordering, depending on whether a user, admin, or security admin is logged in."
> Can you clarify which pages would be visible based on type of user? For example, would a security admin be able to see the Snapshot Admin and Admin Guide pages?

Regular user: Create Project, My Projects, Submissions, About, Feedback, FAQ
Admin: Create Project, My Projects, Submissions, Snapshot Admin, About, Feedback, FAQ, Admin Guide
Security: Create Project, My Projects, Submissions, Snapshot Admin, About, Feedback, FAQ, Admin Guide, Security

It doesn't have to be in that exact order. I'm just listing out the pages. Feel free to reorder them in a more logical way.

I know it seems silly to have the Submissions page there for an Admin, but I think Bonnie wants it there in case admins want to go through the user’s workflow from their own account. I’ll ask her about whether or not admins still need it or if we can remove it.

> "Please reorder the navigation bar, taking into account those three additional pages and the type of logged in user."
> Would this require updating the navigation bar component in our design system? Currently, our design system doesn’t have navigation bars that vary by type of user.

Yes, it would. Please update our design system with this new information.

### Adding new fields

> "In response to [User-facing information about submission status #2195](https://github.com/hackforla/tdm-calculator/issues/2195), stakeholders asked that the fields being added to the Snapshot Admin page via [Design Snapshot Admin and Audit Log pages #1953](https://github.com/hackforla/tdm-calculator/issues/1953) be added to the Submissions page, with the exception of the Created By, Admin Notes, Save, Reset, and Audit Log fields and buttons."
> Just to clarify, by “fields”, you mean table headers like “Date Submitted”?
> Also, can you confirm that these are the fields that will be added to the Submissions page?
> Project
> Level
> DRO
> Staff Assigned
> Assigned Date
> Fees Paid
> Fees Paid Date
> On Hold
> Approval Status
> Certificate of Occupancy Date

Yes, I meant table headers. I updated the issue description with all the fields that need to be included.

> "Please include the fields that are already on the mockup for the Submissions page, i.e. Name, Address, Date Submitted, and Date Modified."
> Is there a specific order the fields should be in, or is this something I should just use my best judgment?

For the sake of consistency, I think we should have it in the same order as on the Snapshot Admin page, but if you have a better idea, feel free to pitch it.

**Comment by greenishviolet** [comment]
### 3/25 Progress

#### Progress
I have not worked on this issue this week

#### Availability
I have around 4 hours to work on this issue

#### ETA
Possibly 1 or 2 more weeks, depending on feedback I receive on the new design next week.

#### Pics
N/A

**Comment by thetanmancan** [comment]
Question for the product team:
- Do admins need access to the Submissions page? They will already have access to the Snapshot Admin page, which is effectively the admin equivalent of the user-facing Submissions page.

**Comment by ExperimentsInHonesty** [comment]
@thetanmancan @greenishviolet I don't think they need it because they are not submitting snapshots.  And if they wanted to test the feature of submitting and seeing a submitted snapshot in their app, they could create a dummy account for fake user and then test it that way.

But having said that, they might still want to see it.  So make sure that the question gets asked in the Stakeholder meeting 2025-04-01

[Slide #2004 in 2025-04-01](https://docs.google.com/presentation/d/1I7j1MTdg5PgNwdr9OvRBjtM8sAKZwtZJYKTqYSEPCBA/edit?slide=id.g183adb1a6d8_0_27#slide=id.g183adb1a6d8_0_27)

**Comment by greenishviolet** [comment]
### 4/1 Progress

#### Progress
I have not done much this week.

I started working on redesigning the navigation bar. I know we agreed on regular font weight for the nav menu items, but since the text is white against a dark background, the menu labels felt hard to read.

![Image](https://github.com/user-attachments/assets/60f66dc8-1b98-4bcf-a1e1-434a03e7d435)

I made initial mockups for the navigation headers (default state).

![Image](https://github.com/user-attachments/assets/44edac33-7209-4ae5-a0ab-8ea276bfbf7f)

#### Blockers
Just a couple questions:
1. Have stakeholders decided if they want the Snapshot pages to appear for Admins?
2. Can I get more context about admins and security admins so I know how to order nav menu items?

#### Availability
I have around 4 hours to work on this issue

#### ETA
Probably 3 more weeks, depending on feedback I receive on the new design next week.

#### Pics
N/A

**Comment by jbethrodriguez** [comment]
Ask stakeholder on 2025-04-01

> Do admins need access to a Submissions page on their accounts?

No, admins do not need access

**Comment by marlenamellody** [comment]
Hi @greenishviolet, thanks for your updates!

- Regarding the nav bar, I definitely understand your point. Currently, on the dev site, the links have normal weight. Based on my quick research on best practices, normal weight is recommended for the default state as it is a familiar standard and provides better legibility in longer menus. Bold weight is typically reserved for active or hover states, as it creates emphasis. However, if you think normal weight for (default) nav links presents an accessibility or readability issue, I’d be happy to discuss it with the team. 

> Can I get more context about admins and security admins so I know how to order nav menu items?

Admins will have two additional pages they can access - the admin guide and the snapshot admin.
Security admins can grant or revoke admin access. 

<details><summary> Here's the current nav order for security admins: </summary>

<img width="1437" alt="Image" src="https://github.com/user-attachments/assets/688313d8-536b-4492-bbc3-ed44612efed4" />
</details>

<details><summary> What security admins are able to access:</summary>

<img width="1440" alt="Image" src="https://github.com/user-attachments/assets/0904ce71-2f6a-4c0a-a2f7-4969dc767aac" />
</details>

Please reach out if you have additional questions or need further clarification.

**Comment by greenishviolet** [comment]
@ExperimentsInHonesty @thetanmancan @marlenamellody—Is it okay to hide comments from before 2/18 (aka. when this issue was about updating tables in the Submit Snapshots Pages)?

**Comment by thetanmancan** [comment]
Hi @greenishviolet. I just hid them for you.

**Comment by greenishviolet** [comment]
Thanks, @thetanmancan!

**Comment by thetanmancan** [comment]
Hi @greenishviolet. There were some changes requested by the stakeholders during the 2025-04-01 meeting. See below.

- [These changes](https://github.com/hackforla/tdm-calculator/issues/1953#issuecomment-2776334734) from #1953 are also being implemented on the Submissions page.
- The "Date Modified" column will be removed and a "Status Updated" column will be added. "Status Updated" shows the most recent date upon which any of the below fields were updated by an admin:
  - Level
  - DRO
  - Staff Assigned
  - Assigned Date
  - Invoice Status
  - Invoice Paid Date
  - On Hold
  - Approval Status
  - Certificate of Occupancy Date

I've already made these changes to this issue's description under the Details section.

I'm so sorry I'm getting these notes to you so late; I've been bogged down in trying to address a lot of other TDM stuff. Thank you for continuing to work on this issue. Please feel free to reach out to me with any questions.

**Comment by ExperimentsInHonesty** [comment]
@thetanmancan @greenishviolet Sorry to bother you both...  Do we think this work will be ready to show design lead by tomorrow 2025-04-09, so that I can see it on 2025-04-10, so that it can go in the 2025-04-15 stakeholder deck.  

If you think the design would be ready for 2025-04-15 but not by this Thursday... Tristan can approve it, when you finish and he can send me a text message for a last min meeting.

**Comment by greenishviolet** [comment]
# 4/8 Update

## Progress

### Submissions Table

@ExperimentsInHonesty @thetanmancan @marlenamellody 
> Do we think this work will be ready to show design lead by tomorrow 2025-04-09, so that I can see it on 2025-04-10, so that it can go in the 2025-04-15 stakeholder deck.

The submissions table is ready to be shared (unless there need to be updates made to it). I referred to Nilakshi's Snapshot Admin and Nami's My Projects tables when building the Submissions table, so there is consistency in the order of fields and sizes/widths of table columns.

I made 2 versions of the table because I wasn't sure if the DRO column should be text-only or if it should be a dropdown like the DRO column on the My Projects Page. I assumed that the other fields in the column should be text only since Admin would update that information.

#### Version 1 - text only
![Image](https://github.com/user-attachments/assets/fcfc023e-f792-40e7-9823-fb27f3ef7f24)

#### Version 2 - DRO with dropdown
![Image](https://github.com/user-attachments/assets/99b61e54-3ee1-4eb9-955e-14aadafb8441)

### Navigation Bar

I am still working on the navigation bar. I did some research on making light text readable against a dark background. It seems that the best option would be to make the text for the primary navigation bold, which is what the LADOT website does.

I am planning on working on hover states for the menu items this week.

#### Navigation Bar - Regular User
![Image](https://github.com/user-attachments/assets/11d26e7f-d46a-41ba-9a69-191c52d66b2e)

- Changed "Hello, HfLA Admin" to "My Account" because while "Hello, HFLA Admin" adds personalization to the website, the label doesn't communicate to users that this is where they should go to update their account information
- Added visual hierarchy by making the font weight in the secondary navigation regular

#### Navigation Bar - Admin
![Image](https://github.com/user-attachments/assets/18b91bb3-c0dd-4341-92a9-87db6d93f15e)

Same changes as the navigation bar for regular users, with the following additions
- Removed the Submissions page
- Reordered the order of the links in the primary navigation so **Snapshot Admin** and **Admin Guide** are shown first and second. This is the order of the links:
  1.  Snapshot Admin
  2. Admin Guide
  3. Create Project
  4. My Projects
  5. About
  6. Feedback
  7. FAQ

#### Navigation Bar - Security Admin
![Image](https://github.com/user-attachments/assets/3cc9ed7a-e2fd-4c79-8f4c-7addeb326c10)

Same changes as the navigation bar for regular users, with the following additions
- Removed the Submissions page
- Reordered the order of the links in the primary navigation so **Snapshot Admin** and **Admin Guide** are shown first and second. This is the order of the links:
  1. Security 
  2. Snapshot Admin
  3. Admin Guide
  4. Create Project
  5. My Projects
  6. About
  7. Feedback
  8. FAQ


## Blockers
I don't think I have any blockers for the Submissions table since the requirements for that part of this issue are very straightforward. However, I would like to get confirmation or feedback on the order of navigation links for Admin and Security Admin.

## Availability
I have around 4 hours to contribute to this issue this week.

## ETA
Possibly 1 or 2 more weeks for the navigation bar. 1 more week for the submissions table.

**Comment by greenishviolet** [comment]
@thetanmancan - I will try to make an updated table before tonight's meeting, but I wanted to respond to your comments.

### Getting rid of the word "Date"
I don't think we should completely get rid of the word "date" because, from an information architecture standpoint, the word helps users build expectations about what kind of data they will find in the column. Since there are so many fields in the table, I think to minimize confusion, labels should be as specific and clear as possible. Without the word "Date," there may be confusion. For example:

- “Date Submitted” to just “Submitted” - This can be misconstrued to "Was this submitted? Yes or no?"
  - An alternative can be "Submitted on"
- “Certificate of Occupancy Date” to just “Certificate of Occupancy” - This is unclear because it sounds like it's asking if the snapshot has certificate of occupancy.
- "Assigned Date” to just “Assigned” - We need to have "Date" in this because right before this column, we have a "Staff Assigned" field
  - Maybe this can be "Assigned on"?

### Moving DRO & Level before Date Submitted
I'm on the fence about this because I'm not quite sure what DRO is. However, I believe that Date Submitted has more importance than the 2 fields because:

1. It gives users additional context about the Snapshot, which may help them remember what Snapshot they submitted.
2. It helps them keep a record of what they have submitted.
  3. This might be a repeat of the first point.

**Comment by greenishviolet** [comment]
### Updated design from 4/9

![Image](https://github.com/user-attachments/assets/63c0f672-9843-4d07-89f8-3a871c05e060)

**Comment by thetanmancan** [comment]
Hi @greenishviolet. Since your design has been approved, can you please take the following actions:
1. Write the navigation bar redesign that you're working on as a separate design issue.
2. Create a development handoff issue for the Submission page.

Thank you.

**Comment by thetanmancan** [comment]
Hi @greenishviolet. I wanted to create a development handoff issue for this, but I'm now realizing that the dropdowns for some of the filters weren't designed. Since those are being designed as part of #1953, do you want to work with @NilakshiS to get those mocked up?

**Comment by ExperimentsInHonesty** [comment]
@thetanmancan

>Hi @greenishviolet. I wanted to create a development handoff issue for this, but I'm now realizing that the dropdowns for some of the filters weren't designed. Since those are being designed as part of https://github.com/hackforla/tdm-calculator/issues/1953, do you want to work with @NilakshiS to get those mocked up?

 I thought we talked about handing this off to dev now.  They need to create the backend database which is why they need the mockup, the drop-downs can be added by Dev on another issue.

**Comment by thetanmancan** [comment]
> [@thetanmancan](https://github.com/thetanmancan)
> 
> > Hi [@greenishviolet](https://github.com/greenishviolet). I wanted to create a development handoff issue for this, but I'm now realizing that the dropdowns for some of the filters weren't designed. Since those are being designed as part of [#1953](https://github.com/hackforla/tdm-calculator/issues/1953), do you want to work with [@NilakshiS](https://github.com/NilakshiS) to get those mocked up?
> 
> I thought we talked about handing this off to dev now. They need to create the backend database which is why they need the mockup, the drop-downs can be added by Dev on another issue.

@ExperimentsInHonesty Yes. I wrote this comment before our meeting.

@greenishviolet, for the sake of expediency, I'll write the handoff isssue for this and then write up a separate design issue to address the other parts of this page. I'll assign it to you to when I've finished with it.

**Comment by thetanmancan** [comment]
@ExperimentsInHonesty The handoff issue has been written as:
- #2372.

**Comment by thetanmancan** [comment]
Hi @greenishviolet. Please move all relevant designs from your 2004 Figma page to a new one for this one I assigned to you:
- #2373 

Once that is done, please complete the rest of the relevant action items on this issue before closing it. Thank you.

**Comment by greenishviolet** [comment]
### 4/15 Progress

#### Progress
- Reduced padding between page header and table
- Updated the column widths and made the values in the Levels column right-aligned
- Added measurements for the table column width and table cell padding for dev
- Annotated text styles just in case
- Marked the design Ready for Dev

#### Pictures

![Image](https://github.com/user-attachments/assets/f8c0b8b3-dd6a-4a0d-a55e-915a769a631c)

**Comment by ExperimentsInHonesty** [comment]
- Created issue for discussing menu update with Stakeholder #2420
